### PR TITLE
chore: Relying on feature flag

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Make sure we check feature flag to expose organizations, also, although trpc checks are there, introduced preemptive checks too. Added feature flag check when showing org profile too.

## Type of change

  - [X] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

1. Try to access org wizard step with a non-org user
2. Try to access wizard with feature flag set to off
3. Try to access org profile with feature flag set to off

## Mandatory Tasks

  - [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Review by ericrommel — DISMISSED** [review]
LGTM, from the UI testing perspective. Code and other things should be checked by the other developers/reviewers

File: apps/web/pages/settings/organizations/[id]/about.tsx [hunk 1]
```diff
@@ -25,6 +29,45 @@ const LayoutWrapper = (page: React.ReactElement) => {
...
+  const membership = await prisma.membership.findFirst({
+    where: {
+      userId: session?.user.id,
+      teamId: session?.user.organizationId,
+    },
+    select: {
+      role: true,
+    },
+  });
+  console.log({ membership });
```
Comment (zomars) [inline apps/web/pages/settings/organizations/[id]/about.tsx:59]:
```suggestion
```

File: apps/web/pages/settings/organizations/[id]/add-teams.tsx [hunk 2]
```diff
@@ -19,6 +23,45 @@ const AddNewTeamsPage = () => {
...
+  const membership = await prisma.membership.findFirst({
+    where: {
+      userId: session?.user.id,
+      teamId: session?.user.organizationId,
+    },
+    select: {
+      role: true,
+    },
+  });
+  console.log({ membership });
```
Comment (zomars) [inline apps/web/pages/settings/organizations/[id]/add-teams.tsx:53]:
```suggestion
```

File: apps/web/pages/settings/organizations/[id]/onboard-admins.tsx [hunk 3]
```diff
@@ -22,6 +26,45 @@ const OnboardTeamMembersPage = () => {
...
+  const membership = await prisma.membership.findFirst({
+    where: {
+      userId: session?.user.id,
+      teamId: session?.user.organizationId,
+    },
+    select: {
+      role: true,
+    },
+  });
+  console.log({ membership });
```
Comment (zomars) [inline apps/web/pages/settings/organizations/[id]/onboard-admins.tsx:56]:
```suggestion
```

File: apps/web/pages/settings/organizations/[id]/set-password.tsx [hunk 4]
```diff
@@ -25,6 +29,45 @@ const LayoutWrapper = (page: React.ReactElement) => {
...
+  const membership = await prisma.membership.findFirst({
+    where: {
+      userId: session?.user.id,
+      teamId: session?.user.organizationId,
+    },
+    select: {
+      role: true,
+    },
+  });
+  console.log({ membership });
```
Comment (zomars) [inline apps/web/pages/settings/organizations/[id]/set-password.tsx:59]:
```suggestion
```

File: apps/web/pages/settings/organizations/[id]/set-password.tsx [hunk 5]
```diff
@@ -25,6 +29,45 @@ const LayoutWrapper = (page: React.ReactElement) => {
   );
 };
 
+export const getServerSideProps = async ({ req, res }: GetServerSidePropsContext) => {
```
Comment (zomars) [inline apps/web/pages/settings/organizations/[id]/set-password.tsx:32]: Can we abstract and reuse this logic?

File: packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx [hunk 6]
```diff
@@ -135,13 +135,12 @@ export const VerifyCodeDialog = ({
   );
 };
 
-export const CreateANewOrganizationForm = () => {
+export const CreateANewOrganizationForm = ({ slug }: { slug?: string }) => {
```
Comment (zomars) [inline packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx:138]: Is this change required?

**Review by zomars — CHANGES_REQUESTED** [review]
Thank you for your contribution @leog Left some things to address first

File: packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx (hunk 6 above)
Comment (leog) [reply packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx:138]: We relied on router query params and now we get it through serverSideProp

File: apps/web/pages/settings/organizations/[id]/set-password.tsx (hunk 5 above)
Comment (leog) [reply apps/web/pages/settings/organizations/[id]/set-password.tsx]: Done, good suggestion

**Review by zomars — APPROVED** [review]
LGTM
